src/webportal/src/app/job-submission/job-submission-page.jsx

```diff
@@ -0,0 +1,435 @@
+            const updated = {...prev};
+            delete updated[id];
+            return updated;
+          }
+        } else {
+          if (prev !== undefined && prev[id] !== msg) {
+            return {
+              ...prev,
+              [id]: msg,
+            };
+          }
+        }
+        return prev;
+      });
+    },
+    [setErrorMessages]
+  );
+
+  const contextValue = useMemo(
+    () => ({
+      vcNames,
+      errorMessages,
+      setErrorMessage,
+    }),
+    [vcNames, errorMessages, setErrorMessage]
+  );
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+    if (params.get('op') === 'resubmit') {
+      const jobName = params.get('jobname') || '';
+      const user = params.get('user') || '';
+      if (user && jobName) {
+        fetchJobConfig(user, jobName)
+          .then((jobConfig) => {
+            const [jobInfo, taskRoles, parameters, , extras] = getJobComponentsFromConfig(
+              jobConfig,
+              {vcNames},
+            );
+            jobInfo.name = generateJobName(jobInfo.name);
+            if (get(jobConfig, 'extras.submitFrom')) {
+              delete jobConfig.extras.submitFrom;
+            }
+            setJobProtocol(new JobProtocol(jobConfig));
+            setJobTaskRoles(taskRoles);
+            setParameters(parameters);
+            setJobInformation(jobInfo);
+            setExtras(extras);
+            setLoading(false);
+          })
+          .catch(alert);
+      }
+    } else {
+      setLoading(false);
+    }
+  }, []);
+
+
+  useEffect(() => {
+    // docker info will be updated in-place
+    const preTaskRoles = JSON.stringify(jobTaskRoles);
+    const taskRolesManager = new TaskRolesManager(jobTaskRoles);
+    taskRolesManager.populateTaskRolesDockerInfo();
+    const [
+      updatedSecrets,
+      isUpdated,
+    ] = taskRolesManager.getUpdatedSecretsAndLinkTaskRoles(secrets);
+
+    const curTaskRoles = JSON.stringify(jobTaskRoles);
+    if (preTaskRoles !== curTaskRoles) {
+      setJobTaskRolesState(jobTaskRoles);
+    }
+
+    if (isUpdated) {
+      setSecrets(updatedSecrets);
+    }
+  }, [jobTaskRoles]);
+
+
+  // update component if yamlText is not null
+  useEffect(() => {
+    if (!isNil(yamlText)) {
+      const updatedJob = JobProtocol.fromYaml(yamlText);
+      if (isNil(updatedJob)) {
+        return;
+      }
+      const [
+        updatedJobInformation,
+        updatedTaskRoles,
+        updatedParameters,
+        updatedSecrets,
+        updatedExtras,
+      ] = getJobComponentsFromConfig(updatedJob, {vcNames});
+      if (extras.tensorBoard) {
+        const updatedTensorBoardExtras = updatedExtras.tensorBoard || {};
+        if (!isValidUpdatedTensorBoardExtras(extras.tensorBoard, updatedTensorBoardExtras)) {
+          updatedExtras.tensorBoard = extras.tensorBoard;
+        }
+      }
+      setJobInformation(updatedJobInformation);
+      setJobTaskRolesState(updatedTaskRoles);
```

Comment:
Add `setProtocol` here. Otherwise we may missing some field. Such as `version`, `contribution`

Reply:
done